### DISCUSSION TITLE
How should we determine future Ichimoku period dates?

### DISCUSSION THREAD

**Discussion by DaveSkender** [DaveSkender/Stock.Indicators#1060] [Q&A]
Ichimoku Cloud currently lacks needed future values beyond the last price (see #675 for problem statement).  The calculation and quantity of future values is known and not an open concern, we know how to do that.  We only need to decide on the best date values to apply to those future dates.

**Comment by DaveSkender** [comment]
There are other options to call third-party APIs to get market and asset calendar info, but we won’t implement those.  We don’t want to introduce external dependencies, and it violates our [guiding principles](https://github.com/DaveSkender/Stock.Indicators/discussions/648) for encapsulation.

**Comment by hieu-n** [comment]
My suggestion would be adding two new properties: `SenkouSpanAFutureList` and `SenkouSpanBFutureList` to `IchimokuResult`.
I suck at naming variables, so someone could come up with better names. The idea is they are simply `List<decimal>` which contain future values for `SenkouSpanA` and `SenkouSpanB`.

Not sure about everyone, but most of time, I only care about the number of bars forward/backward, so the date is not important.
